Traducir.Web.Net/wwwroot/js/string-search.js

```diff
@@ -0,0 +1,94 @@
+function initializeStringSearch () {
+    const queryDropdowns = document.querySelectorAll("select.js-string-query-filter");
+    const queryTextInputs = document.querySelectorAll("input[type=text].js-string-query-filter");
+    const queryLinks = document.querySelectorAll("a.js-string-query-filter");
+    let initialQueryFilters;
+
+    window.onload = function () {
+        initialQueryFilters = clone(stringQueryFilters);
+
+        setInputsFromCurrentFilters();
+        hookDropdowns();
```

Comment:
Whenever the user changes a filter, the new state of all the filters is pushed to the browser history. When the user hits "back", the filters need to be reset to the previous state by using the state object supplied to the `onpopstate` event.

That said, that initial call to `setInputsFromCurrentFilters` from inside `window.onload` is indeed unnecessary since the filters come already with their proper values from the initial page load. It's fixed now.